examples/example_common.h

```diff
@@ -0,0 +1,38 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/*
+ * SplinterDB - Common header file for example programs.
+ */
```

Comment:
Some of the stuff in this file is pillaged from existing splinter header files. We need this so that the example programs do not have to depend on splinterdb `.h` files.